snuba/state/__init__.py

```diff
@@ -13,6 +13,7 @@
 
 from snuba import settings
 from snuba.redis import redis_client as rds
+from snuba.web.query_metadata import SnubaQueryMetadata
```

Comment:
Please move that file here in state, this is a very low level abstraction and related with recording the query stats. Thus, (to make high level sbatrctions depend on lower and not the other way around) it makes sense that snuba.web depends on state while it makes less sense that state depends on snuba.web.
Following this principle when laying down dependencies and placing classes around also reduces the risk you will ever encounter a circular dependency.